### ISSUE TITLE
Feedback Requested: Test Salesforce CLI with Multiple Package Directories for source:deploy and retrieve 

### ISSUE THREAD

**Issue by clairebianchi** [forcedotcom/cli#556]
The Salesforce CLI team has been working hard to add support for multiple packages defined in the same SFDX project [MPD]. Because this is a large change to the way the CLI works with your projects, we'd like to get your feedback along the way as we continue to develop MPD. 

Last month we launched support for source push and pull, [see this issue](https://github.com/forcedotcom/cli/issues/379) where we asked for testers.

The next phase of work is complete, and a test version is available.

This phase is bringing the same support for MPD to the org development commands `source:deploy `and `source:retrieve`. To understand how to set up your project for MPD see our [documentation here](https://developer.salesforce.com/docs/atlas.en-us.226.0.sfdx_dev.meta/sfdx_dev/sfdx_dev_ws_mpd.htm). It is currently written for source:push and pull but the project set up is the same for deploy and retrieve.

Your testing is vital to us insuring that current workflows are not disrupted by this change. We greatly appreciate everyone who tested for push and pull, we could not have launched without your help. We hope you can test again.

 Please install and test out your current workflow to ensure MPD is working as expected.
To install, run 
`sfdx plugins:install salesforce-alm@mpd`  
and verify you have version `49.3.1-4` by running 
`sfdx plugins --core`

Please comment on this issue when you find bugs and create a new issue with [MPD] as the prefix to your issue title.

Thank you in advance for your help

**NOTE: If you install the test version of MPD please remember to uninstall it from your CLI when you are done. You are overwriting the core plug-ins with the MPD branch and will not get updates as they come out.**

To uninstall, run 
`sfdx plugins:uninstall salesforce-alm`

**Comment by honzaluksch** [comment]
I've just tested the update on my multipackage project with following packages:
- core (default) - objects, permission sets, classes, components,..
- portal - objects, permission sets, classes, components,..
- permission-set-groups - permission set groups only

Test scenario: I want to retrieve changes on permission set group "Manager" (part of permission-set-groups package) from sandbox org to my local repository.

### Retrieve:
Running retrieve with source path (-p) worked well.
```
sfdx force:source:retrieve -p sfdx-packages/permission-set-groups/main/default/permissionsetgroups/Manager.permissionsetgroup-meta.xml

=== Retrieved Source
FULL NAME    TYPE                PROJECT PATH
───────────  ──────────────────  ────────────────────────────────────────────────────────────────────────────────────────────────────────────
Manager  PermissionSetGroup  sfdx-packages/permission-set-groups/main/default/permissionsetgroups/Manager.permissionsetgroup-meta.xml
```

However, running similar operation with metadata option (-m) retrieved the source file twice - into correct package + into default package as new file.
```
sfdx force:source:retrieve -m PermissionSetGroup:Manager

=== Retrieved Source
FULL NAME    TYPE                PROJECT PATH
───────────  ──────────────────  ────────────────────────────────────────────────────────────────────────────────────────────────────────────
Manager  PermissionSetGroup  sfdx-packages/core/main/default/permissionsetgroups/Manager.permissionsetgroup-meta.xml
Manager  PermissionSetGroup  sfdx-packages/permission-set-groups/main/default/permissionsetgroups/Manager.permissionsetgroup-meta.xml
```


### Deploy:
Deploy worked well using both source path (-p) and metadata option (-p).
```
sfdx force:source:deploy -p sfdx-packages/permission-set-groups/main/default/permissionsetgroups/Manager.permissionsetgroup-meta.xml

=== Deployed Source
FULL NAME    TYPE                PROJECT PATH
───────────  ──────────────────  ────────────────────────────────────────────────────────────────────────────────────────────────────────────
Manager  PermissionSetGroup  sfdx-packages/permission-set-groups/main/default/permissionsetgroups/Manager.permissionsetgroup-meta.xml
```


```
sfdx force:source:deploy -m PermissionSetGroup:Manager`

=== Deployed Source
FULL NAME    TYPE                PROJECT PATH
───────────  ──────────────────  ────────────────────────────────────────────────────────────────────────────────────────────────────────────
Manager  PermissionSetGroup  sfdx-packages/permission-set-groups/main/default/permissionsetgroups/Manager.permissionsetgroup-meta.xml
```

**Comment by nramyasri-sf** [comment]
@honzaluksch , is this only happening with PermissionSetGroup metadata type? Is the behavior something you are consistently seeing?

**Comment by Gurudata** [comment]
**sfdx force:source:deploy commands deploys packages by following the sequence of packages mentioned in the root directory and not the sequence mentioned in sfdx-project.json**

Steps To Reproduce:

I have a below folder structure where Package2 comes before Package1
PackageDirectory
---Package2
---Package1

My sfdx-project.json has the following sequence where Package1 comes before Package2
Package1
Package2

When I run the below command to deploy multiple packages, Package2 is deployed before Package1. The sequence in sfdx-project.json is ignored.
`sfdx force:source:deploy -p PackageDirectory -u devOrg
`
**Expected result**
sfdx force:source:deploy should respect package sequence in sfdx-project.json, Package1 should be deployed before Package2

**Actual result**
sfdx force:source:deploy follows the sequence of packages in the root package directory and not the sequence in sfdx-project.json

**Comment by nramyasri-sf** [comment]
@Gurudata , thanks for reporting this. We will provide an update once this has been fixed.

**Comment by amphro** [comment]
This has now been released in `49.10.0`
